mealie/services/migrations/cookn.py

```diff
@@ -0,0 +1,409 @@
+
+            if num_match:
+                # If there's a current step, store it before starting a new one
+                if current_step:
+                    steps.append("\n".join(current_step))
+                    current_step = []
+
+                current_step.append(num_match.group(2))
+            elif bullet_match:
+                if current_step:
+                    steps.append("\n".join(current_step))
+                    current_step = []
+
+                current_step.append(bullet_match.group(1))
+            else:
+                # Continuation of a previous step
+                if current_step:
+                    current_step.append(line)
+                else:
+                    # If no clear separator is found, treat each new line as a new step
+                    steps.append(line)
+
+        if current_step:
+            steps.append(" ".join(current_step))
+
+        return steps
+
+    def _process_recipe_document(
+        self, _recipe_row: dict[str, Any], db: DSVParser
+    ) -> dict[str, str | list[str] | list[RecipeIngredient]]:
+        """Parses recipe row from the Cook'n recipe table."""
+        recipe_data: dict[str, str | list[str] | list[RecipeIngredient]] = {}
+
+        # Select db values
+        _recipe_id = db.get_data(_recipe_row, "ID")
+        _recipe_desc_row = db.query_by_id("temp_recipe_desc", "ID", [_recipe_id])[0]
+        _chapter_id = db.get_data(_recipe_desc_row, "PARENT")
+        _chapter_row = db.query_by_id("temp_chapter_desc", "ID", [_chapter_id])[0]
+        _cookbook_id = db.get_data(_chapter_row, "PARENT")
+        _cookbook_row = db.query_by_id("temp_cookBook_desc", "ID", [_cookbook_id])[0]
+        _media_row = db.query_by_id("temp_media", "ENTITY_ID", [_recipe_id])[0]
+
+        # Parse general recipe info
+        cookbook = db.get_data(_cookbook_row, "TITLE")
+        chapter = db.get_data(_chapter_row, "TITLE")
+        name = db.get_data(_recipe_desc_row, "TITLE")
+        description = db.get_data(_recipe_desc_row, "DESCRIPTION")
+        serves = db.get_data(_recipe_row, "SERVES")
+        # yields = db.get_data(_recipe_row, "YIELD")
+        prep_time = int(db.get_data(_recipe_row, "PREPTIME"))
+        cook_time = int(db.get_data(_recipe_row, "COOKTIME"))
+
+        recipe_data["recipeCategory"] = [cookbook + " - " + chapter]
+        recipe_data["name"] = name
+        recipe_data["description"] = description
+        recipe_data["recipeYield"] = serves
+        recipe_data["prepTime"] = format_time(prep_time)
+        recipe_data["performTime"] = format_time(cook_time)
+        recipe_data["totalTime"] = format_time(prep_time + cook_time)
+
+        # Parse image file
+        image_path = self._parse_media(_cookbook_id, _chapter_id, _recipe_id, db)
+        if image_path is not None:
+            recipe_data["image"] = [image_path]
+
+        # Parse ingredients
+        recipe_data["ingredients"] = self._parse_ingredients(_recipe_id, db)
```

Comment:
Would we be able to rename this so something more explicitly intermittent, such as `"_parsed_ingredients"`? I saw your earlier comment on why you did it this way, but at first glance it's confusing and looks like an error.

(The core issue is more of a problem with how other migrations work which results in data loss, but that's a task for a separate PR).